### ISSUE TITLE
Pool connection limits are corrupted when all pools are unhealthy

### ISSUE THREAD

**Issue by simonratner** [neumino/rethinkdbdash#85]
When all pools become unhealthy, with discovery=false, the calculation to spread/trim the connection limits between remaining pools is incorrect, and can result in pretty bad corruption of the pools (e.g. zero or negative limits when the pool becomes healthy again).

The problem seems to be when removing the last pool from the healthy set, or adding the first pool into an empty healthy set, there is nowhere to spread/trim the limits from, and the delta is lost. 

Additionally, integer rounding error accumulate over multiple healthy/unhealthy cycles, and that can also cause the numbers to diverge significantly.

Here's a script I used to test:
```js
var r = require('rethinkdbdash')({  // rethinkdbdash 2.0.3
    buffer: 4,
    max: 8,
    pool: true,
    silent: true,
    discovery: false,
    servers: [
        { host: 'localhost', port: 28015 },
        { host: 'localhost', port: 28016 }
    ]});

r.getPoolMaster().getPools().forEach(function(pool) {
    ['created-connection', 'empty', 'not-empty'].forEach(function(ev) {
        pool.on(ev, function() {
            console.log(ev + ' [' + this.id + '] buffer=' + this.options.buffer + ', max=' + this.options.max);
        });
    });
});
r.getPoolMaster().on('size', function(num) {
    if (num == 0) {
        console.log('[all pool empty]');
    }
});

var query = function() {
    r.db('rethinkdb').table('server_status').then(function(servers) {
        var pools = r.getPoolMaster().getPools().map(function(pool) {
            return '[' + pool.id + '] buffer=' + pool.options.buffer + ', max=' + pool.options.max;
        });
        console.log('ok', pools.join(', '));
    }).catch(function(err) {
        console.log(err.name);
    });
};

setInterval(query, 2000);
query();
```
I then randomly killed and restarted the servers a few times, including a period when both servers were down. Here's some sample output; you can see by the end that connection limits are not very happy:

```
not-empty [80290] buffer=2, max=4
not-empty [5322] buffer=2, max=4
created-connection [80290] buffer=2, max=4
created-connection [80290] buffer=2, max=4
ok [80290] buffer=2, max=4, [5322] buffer=2, max=4
created-connection [80290] buffer=2, max=4
created-connection [5322] buffer=2, max=4
created-connection [5322] buffer=2, max=4
created-connection [5322] buffer=2, max=4
created-connection [80290] buffer=2, max=4
ok [80290] buffer=2, max=4, [5322] buffer=2, max=4
ok [80290] buffer=2, max=4, [5322] buffer=2, max=4
empty [80290] buffer=2, max=4
[all pool empty]
empty [5322] buffer=4, max=8
ReqlDriverError
not-empty [80290] buffer=2, max=4
[all pool empty]
empty [80290] buffer=2, max=4
not-empty [5322] buffer=4, max=8
[all pool empty]
empty [5322] buffer=4, max=8
ReqlDriverError
ReqlDriverError
not-empty [80290] buffer=2, max=4
[all pool empty]
empty [80290] buffer=2, max=4
not-empty [5322] buffer=4, max=8
[all pool empty]
empty [5322] buffer=4, max=8
ReqlDriverError
ReqlDriverError
ReqlDriverError
ReqlDriverError
not-empty [80290] buffer=2, max=4
[all pool empty]
empty [80290] buffer=2, max=4
not-empty [5322] buffer=4, max=8
[all pool empty]
empty [5322] buffer=4, max=8
ReqlDriverError
ReqlDriverError
ReqlDriverError
ReqlDriverError
ReqlDriverError
ReqlDriverError
ReqlDriverError
ReqlDriverError
not-empty [80290] buffer=2, max=4
created-connection [80290] buffer=2, max=4
created-connection [80290] buffer=2, max=4
created-connection [80290] buffer=2, max=4
not-empty [5322] buffer=4, max=8
created-connection [5322] buffer=4, max=8
created-connection [5322] buffer=4, max=8
created-connection [5322] buffer=4, max=8
created-connection [5322] buffer=4, max=8
created-connection [5322] buffer=4, max=8
ok [80290] buffer=-2, max=-4, [5322] buffer=4, max=8
ok [80290] buffer=-2, max=-4, [5322] buffer=4, max=8
ok [80290] buffer=-2, max=-4, [5322] buffer=4, max=8
ok [80290] buffer=-2, max=-4, [5322] buffer=4, max=8
ok [80290] buffer=-2, max=-4, [5322] buffer=4, max=8
```

**Comment by neumino** [comment]
Arg I wrote a fix yesterday on my laptop but forgot to merge it on my desktop.
I'll push it after dinner :)

Thanks @simonratner for the report

**Comment by simonratner** [comment]
:+1:

**Comment by neumino** [comment]
Released in 2.0.5

Thanks again @simonratner :)
